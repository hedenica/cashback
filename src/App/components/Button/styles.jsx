import styled from 'styled-components';
import { shade } from 'polished';

export const Container = styled.button`
  background: #fbab8c;
  height: 56px;
  border-radius: 10px;
  border: 0;
  padding: 0 16px;
  color: #666360;
  width: 100%;
  font-weight: 500;
  margin-top: 16px;
  transition: background-color 0.5s;

  &:hover {
    background: ${shade(0.2, '#fbab8c')};
  }
`;
