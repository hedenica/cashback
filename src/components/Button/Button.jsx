import React from 'react';
import PropTypes from 'prop-types';

import { Container } from './styles';

const Button = ({ children, ...rest }) => (
  <Container type="button" {...rest}>
    {children}
  </Container>
);

export default Button;

Button.propTypes = {
  children: PropTypes.node.isRequired,
};
