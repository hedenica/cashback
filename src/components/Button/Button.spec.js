import React from 'react';
import { render } from  '@testing-library/react'

import Button from './Button'

describe('<Button />', () => {
  it ('should match the button snapshot', () => {
    const { container } = render(
      <Button type="button">
        Clique aqui
      </Button>
    )
    expect(container).toMatchSnapshot();
  })
});

