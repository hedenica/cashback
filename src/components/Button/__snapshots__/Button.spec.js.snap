// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match the button snapshot 1`] = `
<div>
  <button
    class="sc-bdfBwQ ifSYyB"
    type="button"
  >
    Clique aqui
  </button>
</div>
`;
